lib/iris/tests/unit/coords/test_Coord.py

```diff
@@ -332,7 +332,9 @@ def test_dim_1d(self):
         )
         for units in ["unknown", "no_unit", 1, "K"]:
             coord.units = units
-            collapsed_coord = coord.collapsed()
+            with mock.patch("warnings.warn") as warn:
```

Comment:
For warnings testing, I think it is more obvious to use [assertWarnsRegex/assertWarnsRegex](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns) or (in Iris, at least) [assertNoWarnsRegex](https://github.com/SciTools/iris/blob/main/lib/iris/tests/__init__.py#L642-L650)